exporters/prometheus/src/exporter_utils.cc

```diff
@@ -242,40 +251,63 @@ void PrometheusExporterUtils::SetData(std::vector<T> values,
  * Histogram => Prometheus Histogram
  */
 template <typename T>
-void PrometheusExporterUtils::SetData(std::vector<T> values,
-                                      const std::vector<double> &boundaries,
-                                      const std::vector<uint64_t> &counts,
-                                      const metric_sdk::PointAttributes &labels,
-                                      std::chrono::nanoseconds time,
-                                      prometheus_client::MetricFamily *metric_family)
+void PrometheusExporterUtils::SetData(
+    std::vector<T> values,
+    const std::vector<double> &boundaries,
+    const std::vector<uint64_t> &counts,
+    const metric_sdk::PointAttributes &labels,
+    const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope,
+    std::chrono::nanoseconds time,
+    prometheus_client::MetricFamily *metric_family)
 {
   metric_family->metric.emplace_back();
   prometheus_client::ClientMetric &metric = metric_family->metric.back();
-  SetMetricBasic(metric, time, labels);
+  SetMetricBasic(metric, time, labels, scope);
   SetValue(values, boundaries, counts, &metric);
 }
 
 /**
  * Set time and labels to metric data
  */
-void PrometheusExporterUtils::SetMetricBasic(prometheus_client::ClientMetric &metric,
-                                             std::chrono::nanoseconds time,
-                                             const metric_sdk::PointAttributes &labels)
+void PrometheusExporterUtils::SetMetricBasic(
+    prometheus_client::ClientMetric &metric,
+    std::chrono::nanoseconds time,
+    const metric_sdk::PointAttributes &labels,
+    const opentelemetry::sdk::instrumentationscope::InstrumentationScope *scope)
 {
   metric.timestamp_ms = time.count() / 1000000;
 
-  // auto label_pairs = ParseLabel(labels);
   if (!labels.empty())
   {
-    metric.label.resize(labels.size());
-    size_t i = 0;
+    metric.label.reserve(labels.size() + 2);
```

Comment:
This should go outside the labels check. In fact, we should not even check if labels is empty. It would just skip the loop.

Reply:
done